### PR TITLE
cmake, doc: Update `test/README.md` and other docs

### PR DESCRIPTION
**Pull request by hebasto**

### CODE REVIEW

**Comment by hebasto** [comment]
cc @maflcko

File: test/README.md [hunk 1]
```diff
@@ -45,54 +47,54 @@ set PYTHONUTF8=1
...
 ```
-test/functional/test_runner.py <testname1> <testname2> <testname3> ...
+build/test/functional/test_runner.py <testname1> <testname2> <testname3> ...
 ```
 
 Wildcard test names can be passed, if the paths are coherent and the test runner
 is called from a `bash` shell or similar that does the globbing. For example,
 to run all the wallet tests:
 
 ```
```
Comment (paplorinc) [inline test/README.md:69]: nit:
\```bash

File: test/README.md [hunk 2]
```diff
@@ -45,54 +47,54 @@ set PYTHONUTF8=1
...
-test/functional/test_runner.py test/functional/wallet*
-functional/test_runner.py functional/wallet* (called from the test/ directory)
-test_runner.py wallet* (called from the test/functional/ directory)
+build/test/functional/test_runner.py test/functional/wallet*
+functional/test_runner.py functional/wallet*  # (called from the build/test/ directory)
```
Comment (paplorinc) [inline test/README.md:71]: nit: the command line usually doesn't interpret these as comments

File: contrib/guix/README.md [hunk 3]
```diff
@@ -261,6 +261,7 @@ details.
 
   - `guix` build commands as in `guix shell --cores="$JOBS"`
   - `make` as in `make --jobs="$JOBS"`
+  - `cmake` as in `cmake --build build -j "$JOBS"`
```
Comment (paplorinc) [inline contrib/guix/README.md:264]: do we need both in guix?

**Review by paplorinc — APPROVED** [review]
ACK e16f24e59f6f88abbec4a79d2c8e88938bbcbf72
Left some nits and questions

File: contrib/guix/README.md (hunk 3 above)
Comment (hebasto) [reply contrib/guix/README.md:264]: GNU Make is still used to build depends.

File: test/README.md (hunk 2 above)
Comment (hebasto) [reply test/README.md:71]: Bash does doesn't it?

Comment (paplorinc) [reply test/README.md:71]: The latest version seems to handle it indeed:

> bash-3.2$ echo 12 #34
 12

but with zsh it's
> echo 12 #34
12 #34

Comment (hebasto) [reply test/README.md:71]: The same Bash style for comments is used in other docs. So, it can be improved in a separated PR.

**Pull request merged**